### ISSUE TITLE
ver. 20.5.0 (205000) don't work

### ISSUE THREAD

**Issue by reno-a** [jitsi/jitsi-meet#8141]
Hello,

F-Droid repos.
The version 20.5.0 (205000) blocked on logo.
I must return to the version 20.4.2 (204020).

Material: Motorola G6 Play / Android 9

**Comment by gitnard** [comment]
Ditto F-Droid repo ver 20.5.0 fails, ver 20.4.2 works fine.
On Samsung Galaxy S5  Android 6.0.1 and on Samsung Tab S2 9.7  Android 7.0.

**Comment by CryptOwl** [comment]
Confirmed using 20.5.0 via F-Droid on a OnePlus 3T with LineageOS 17.1 and latest updates. Full reinstallation of the app did not help.

**Comment by oczkoisse** [comment]
Experiencing the same issue on Oneplus 6T using F-Droid repo and app version 20.5.0. Clearing cache and storage hasn't helped. Tried reinstalling, but that didn't work either.

**Comment by saghul** [comment]
Can you please open an issue on the F-Droid repo? I try to help when requested, but ultimately we are not doing the F-Droid builds.

**Comment by reno-a** [comment]
> Can you please open an issue on the F-Droid repo? I try to help when requested, but ultimately we are not doing the F-Droid builds.

On the F-Droid forum: https://forum.f-droid.org/t/jitsi-meet-ver-20-5-0-205000-dont-work/11747

**Comment by licaon-kter** [comment]
I've disabled the build entry earlier, it will disappear at the next of the current cycle.

I'll try to grab some logs...

**Comment by licaon-kter** [comment]
Logcat says:
```
I ActivityTaskManager: START u0 {act=android.intent.action.MAIN cat=[android.intent.category.LAUNCHER] flg=0x10200000 cmp=org.jitsi.meet/.MainActivity bnds=[50,2050][171,2171]} from uid 10266
I ActivityManager: Start proc 16996:org.jitsi.meet/u0a339 for activity {org.jitsi.meet/org.jitsi.meet.MainActivity}
E org.jitsi.meet: Not starting debugger since process cannot load the jdwp agent.
I org.jitsi.meet: Starting a blocking GC Alloc
I org.jitsi.meet: Starting a blocking GC Alloc
D SoLoader: adding application source: com.facebook.soloader.c[root = /data/app/org.jitsi.meet-QZda_j4kxF8AuUWJL5JNxA==/lib/arm64 flags = 0]
D SoLoader: adding backup source from : com.facebook.soloader.a[root = /data/data/org.jitsi.meet/lib-main flags = 1]
D SoLoader: Preparing SO source: com.facebook.soloader.c[root = /data/app/org.jitsi.meet-QZda_j4kxF8AuUWJL5JNxA==/lib/arm64 flags = 0]
D SoLoader: Preparing SO source: com.facebook.soloader.a[root = /data/data/org.jitsi.meet/lib-main flags = 1]
V fb-UnpackingSoSource: locked dso store /data/user/0/org.jitsi.meet/lib-main
V fb-UnpackingSoSource: releasing dso store lock for /data/user/0/org.jitsi.meet/lib-main (from syncer thread)
V fb-UnpackingSoSource: not releasing dso store lock for /data/user/0/org.jitsi.meet/lib-main (syncer thread started)
D SoLoader: libreactnativejni.so found on /data/data/org.jitsi.meet/lib-main
D SoLoader: libglog_init.so found on /data/data/org.jitsi.meet/lib-main
D SoLoader: libglog.so found on /data/data/org.jitsi.meet/lib-main
D SoLoader: libc++_shared.so found on /data/data/org.jitsi.meet/lib-main
D SoLoader: libc.so not found on /data/data/org.jitsi.meet/lib-main
D SoLoader: libc.so not found on /data/app/org.jitsi.meet-QZda_j4kxF8AuUWJL5JNxA==/lib/arm64
D SoLoader: libdl.so not found on /data/data/org.jitsi.meet/lib-main
D SoLoader: libdl.so not found on /data/app/org.jitsi.meet-QZda_j4kxF8AuUWJL5JNxA==/lib/arm64
D SoLoader: libm.so not found on /data/data/org.jitsi.meet/lib-main
D SoLoader: libm.so not found on /data/app/org.jitsi.meet-QZda_j4kxF8AuUWJL5JNxA==/lib/arm64
D SoLoader: liblog.so not found on /data/data/org.jitsi.meet/lib-main
D SoLoader: liblog.so not found on /data/app/org.jitsi.meet-QZda_j4kxF8AuUWJL5JNxA==/lib/arm64
D SoLoader: libyoga.so found on /data/data/org.jitsi.meet/lib-main
D SoLoader: libfb.so found on /data/data/org.jitsi.meet/lib-main
D SoLoader: libandroid.so not found on /data/data/org.jitsi.meet/lib-main
D SoLoader: libandroid.so not found on /data/app/org.jitsi.meet-QZda_j4kxF8AuUWJL5JNxA==/lib/arm64
D SoLoader: libfb.so found on /data/data/org.jitsi.meet/lib-main
I org.jitsi.meet: Thread[17,tid=17026,Native,Thread*=0x7078b6fc00,peer=0x12e0b5f8,"create_react_context"] recursive attempt to load library "/data/user/0/org.jitsi.meet/lib-main/libfb.so"
D SoLoader: libjsinspector.so found on /data/data/org.jitsi.meet/lib-main
D SoLoader: libfolly_json.so found on /data/data/org.jitsi.meet/lib-main
D SoLoader: libjscexecutor.so found on /data/data/org.jitsi.meet/lib-main
D SoLoader: libjsc.so found on /data/data/org.jitsi.meet/lib-main
D SoLoader: libicu_common.so found on /data/data/org.jitsi.meet/lib-main
D SoLoader: libgnustl_shared.so not found on /data/data/org.jitsi.meet/lib-main
D SoLoader: libgnustl_shared.so not found on /data/app/org.jitsi.meet-QZda_j4kxF8AuUWJL5JNxA==/lib/arm64
```
...so we are missing, some...lib :+1: 

Quick question, why did the app just jump twice in size? The screenshare thing?

**Comment by saghul** [comment]
> Quick question, why did the app just jump twice in size? The screenshare thing?

No, that might be Hermes is not getting stripped?

**Comment by saghul** [comment]
I'll try to take a look.

**Comment by saghul** [comment]
I've found the problem, I'll have a fix hopefully today.

**Comment by saghul** [comment]
This was an F-Droid build issue, fix here: https://gitlab.com/fdroid/fdroiddata/-/merge_requests/7774 I expect it to land shortly and 20.5.0 being available soon after that.

**Comment by 532910** [comment]
2.6.0 is still 80+ Mb, should another issue be opened for this?

**Comment by licaon-kter** [comment]
F-Droid packages the universal build with all arches. It's not an issue.

But, I guess we should pursue a per arch APK.

**Comment by saghul** [comment]
Forgot to follow-up on this. Do you have NDK installed? Looks like the strip step is skipped otherwise, resulting in large APK sizes.

**Comment by licaon-kter** [comment]
Interesting, if the recipe does not request explicitely an NDK version it should have r12b.

JM recipe does not.

Is the presence of say NDK21b enough for the strip step or something else needs to be added recipe-wise?

**Comment by saghul** [comment]
I think that should be enough, but please double-check.

**Comment by licaon-kter** [comment]
@saghul aaaaand, no difference

Any other magic that needs to be invoked?

I mean, your local universal APKs are way smaller that 84000000bytes?

**Comment by saghul** [comment]
> I mean, your local universal APKs are way smaller that 84000000bytes?

Actually no, 88.12 MB is the last beta APK size. Given it includes 4 architectures, it's ~22 each, so pretty good.

**Comment by licaon-kter** [comment]
Ok, so stripping is not the issue.

It the per arch build easy to achieve? Eg. `export ARCH=some` then build as usual

**Comment by 532910** [comment]
[20.4.2](https://f-droid.org/repo/org.jitsi.meet_204020.apk) is also multiarch, but it less than 30Mb

**Comment by licaon-kter** [comment]
@532910 yes, there was a framework change between hermes and jsc, iirc
